pkg/odo/cli/service/backends.go

```diff
@@ -0,0 +1,422 @@
+	}
+}
+
+func (b *OperatorBackend) RunServiceCreate(o *CreateOptions) (err error) {
+	s := &log.Status{}
+
+	// in case of an Operator backed service, name of the service is
+	// provided by the yaml specification in alm-examples. It might also
+	// happen that a user wants to spin up Service Catalog based service in
+	// spite of having 4.x cluster mode but we're not supporting
+	// interacting with both Operator Hub and Service Catalog on 4.x. So
+	// the user won't get to see service name in the log message
+	if !o.DryRun {
+		log.Infof("Deploying service %q of type: %q", o.ServiceName, b.CustomResource)
+		s = log.Spinner("Deploying service")
+		defer s.End(false)
+	}
+
+	// if cluster has resources of type CSV and o.CustomResource is not
+	// empty, we're expected to create an Operator backed service
+	if o.DryRun {
+		// if it's dry run, only print the alm-example (o.CustomResourceDefinition) and exit
+		jsonCR, err := json.MarshalIndent(b.CustomResourceDefinition, "", "  ")
+		if err != nil {
+			return err
+		}
+
+		// convert json to yaml
+		yamlCR, err := yaml.JSONToYAML(jsonCR)
+		if err != nil {
+			return err
+		}
+
+		log.Info(string(yamlCR))
+
+		return nil
+	} else {
+		err = svc.CreateOperatorService(o.KClient, o.EnvSpecificInfo, o.ServiceName, b.group, b.version, b.resource, b.CustomResourceDefinition)
+		if err != nil {
+			// TODO: logic to remove CRD info from devfile because service creation failed.
+			return err
+		} else {
+			s.End(true)
+			log.Successf(`Service %q was created`, o.ServiceName)
+		}
+
+		crdYaml, err := yaml.Marshal(b.CustomResourceDefinition)
+		if err != nil {
+			return err
+		}
+
+		err = svc.AddKubernetesComponentToDevfile(string(crdYaml), o.ServiceName, o.EnvSpecificInfo.GetDevfileObj())
+		if err != nil {
+			return err
+		}
+	}
+	s.End(true)
+
+	return
+}
+
+func (b *OperatorBackend) CompleteServiceDelete(o *DeleteOptions, cmd *cobra.Command, args []string) (err error) {
+	return
+}
+
+func (b *OperatorBackend) ServiceExists(o *DeleteOptions) (bool, error) {
+	return svc.OperatorSvcExists(o.KClient, o.serviceName)
+}
+
+func (b *OperatorBackend) DeleteService(o *DeleteOptions, name string, application string) error {
+	err := svc.DeleteOperatorService(o.KClient, o.serviceName)
+	if err != nil {
+		return err
+	}
+
+	// "name" is of the form CR-Name/Instance-Name so we split it
+	// we ignore the error because the function used below is called in the call to "DeleteOperatorService" above.
+	_, instanceName, _ := svc.SplitServiceKindName(name)
+
+	err = svc.DeleteKubernetesComponentFromDevfile(instanceName, o.EnvSpecificInfo.GetDevfileObj())
+	if err != nil {
+		return errors.Wrap(err, "failed to delete service from the devfile")
+	}
+
+	return nil
+}
+
+func (b *ServiceCatalogBackend) CompleteServiceDelete(o *DeleteOptions, cmd *cobra.Command, args []string) (err error) {
```

Comment:
Might be a good idea to separate the service catalog code into a separate file

Reply:
Makes sense. The file is becoming a mess.

Reply:
Separated into different files.